### PR TITLE
Auto build and CI for windows and linux

### PR DESCRIPTION
**Pull request by xribene**

### CODE REVIEW

File: .github/workflows/cmake_ctest.yml [hunk 1]
```diff
@@ -119,14 +119,31 @@ jobs:
...
+      if: matrix.name == 'Windows'
+      shell: bash
+      run: |
+        echo "STANDALONE_PATH=${{ env.PROJECT_NAME }}_artefacts/${{ env.BUILD_TYPE }}/${{ env.PROJECT_NAME }}.exe" >> $GITHUB_ENV
+
+    - name: Setup Environment Variables for Linux
+      if: matrix.name == 'Linux'
+      shell: bash
+      run: |
+        echo "STANDALONE_PATH=${{ env.PROJECT_NAME }}_artefacts/${{ env.BUILD_TYPE }}/${{ env.PROJECT_NAME }}" >> $GITHUB_ENV
```
Comment (cwitkowitz) [inline .github/workflows/cmake_ctest.yml:145]: Should there be any sort of extension for Linux? I know functionally it shouldn't matter.

File: README.md [hunk 2]
```diff
@@ -144,9 +145,17 @@ cmake .. DCMAKE_OSX_ARCHITECTURES=x86_64
 ```
 
 ## Windows
-HARP has been tested on Windows 10 x64. You can checkout the `windowsBuild` branch and follow the instructions in the README.
+**Configure**
```
Comment (cwitkowitz) [inline README.md:148]: This could be combined with above "Configure".

File: .github/workflows/cmake_ctest.yml (hunk 1 above)
Comment (xribene) [reply .github/workflows/cmake_ctest.yml:145]: I think in Linux it is most common to not have any extension for executable files, so maybe this is fine.

**Comment by cwitkowitz** [comment]
Looks good to me. It builds correctly on Ubuntu 22.04 and macOS 10.15. There's just the one issue with the ZIP for Linux, but I think we can revisit that later.

File: .github/workflows/cmake_ctest.yml [hunk 3]
```diff
@@ -159,26 +182,37 @@ jobs:
...
-      shell: bash
-      run: |
-        iscc "packaging\installer.iss"
-        mv "packaging/Output/${{ env.PRODUCT_NAME }}.exe" "${{ env.ARTIFACTS_PATH }}/"
-            dotnet tool install --global AzureSignTool 
-            AzureSignTool sign -kvu "${{ secrets.AZURE_KEY_VAULT_URI }}" -kvi "${{ secrets.AZURE_CLIENT_ID }}" -kvt "${{ secrets.AZURE_TENANT_ID }}" -kvs "${{ secrets.AZURE_CLIENT_SECRET }}" -kvc ${{ secrets.AZURE_CERT_NAME }} -tr http://timestamp.digicert.com -v "${{ env.ARTIFACTS_PATH }}/${{ env.PRODUCT_NAME }}.exe"
-
-    - name: Upload Exe (Windows)
+      working-directory: ${{ env.ARTIFACTS_PATH }}
+      run: Compress-Archive -Path ./* -DestinationPath ${{ env.PRODUCT_NAME }}.zip
```
Comment (cwitkowitz) [inline .github/workflows/cmake_ctest.yml:193]: Is there any point to zipping the exe file?

File: README.md [hunk 4]
```diff
@@ -177,27 +192,26 @@ After running `package.sh`, you should have a signed and notarized dmg file in t
...
     "configurations": [
         {
-            "name": "(lldb) Launch",
+            "name": "(lldb) Standalone",
             "type": "cppdbg",
             "request": "launch",
-            "program": "${workspaceFolder}/build/HARP_artefacts/Debug/HARP.app",
-            "args": ["./test.wav"], // TODO - remove?
-            "stopAtEntry": false,
+            "program": "${workspaceFolder}/build/HARP_artefacts/Debug/HARP.app", // for macOS
```
Comment (cwitkowitz) [inline README.md:209]: Comments look perfectly fine in VS Code. I'm not sure why they are like this here.

**Review by hugofloresgarcia — APPROVED** [review]
hell yea!

**Pull request merged**